### PR TITLE
 *: implement web sql and web parse commands 

### PR DESCRIPTION
**Pull request by erizocosmico**
Depends on #9 

There's an issue with bblfsh/web. Since it connects to bblfsh immediately we don't have a chance to add the container to the network so they don't see each other. The solution for this is to just defer the connection until it's required. I will make a PR to bblfsh/web and when this is fixed and released we can drop the use of a link.

See: https://github.com/bblfsh/web/pull/153

### CODE REVIEW

File: cmd/srcd-server/engine/web.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+		if err := docker.Kill(gitbaseWebName); err != nil {
+			return err
+		}
+	}
+
+	return Run(Component{
+		Name:  gitbaseWebName,
+		Start: createGitbaseWeb,
+		Options: []docker.ConfigOption{
+			docker.WithPort(port, bblfshWebPrivatePort),
```
Comment (campoy) [inline cmd/srcd-server/engine/web.go]: this should be gitbaseWebPrivatePort

File: cmd/srcd-server/engine/web.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	config := &container.Config{
+		Image: gitbaseWebImage,
+		Env: []string{
+			fmt.Sprintf("GITBASEPG_DB_CONNECTION=root@tcp(%s)/none?maxAllowedPacket=4194304", gitbaseName),
+			fmt.Sprintf("GITBASEPG_BBLFSH_SERVER_URL=%s:%d", bblfshdName, bblfshParsePort),
+		},
```
Comment (campoy) [inline cmd/srcd-server/engine/web.go]: missing `fmt.Sprintf("PORT=%d", gitbaseWebPrivatePort),`

File: cmd/srcd/cmd/web.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		if err != nil {
+			cancel()
+			logrus.Fatalf("could not stop gitbase web at port %d: %v", port, err)
+		}
+	},
+}
+
+var webParseCmd = &cobra.Command{
+	Use:   "parse",
+	Short: "Start bblfsh web",
```
Comment (campoy) [inline cmd/srcd/cmd/web.go]: Start bblfsh web client.

File: cmd/srcd/cmd/web.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+var webCmd = &cobra.Command{
+	Use:   "web",
+	Short: "Start web interfaces for source{d} tools",
+}
+
+var webSQLCmd = &cobra.Command{
+	Use:   "sql",
+	Short: "Start gitbase web",
+	Run: func(cmd *cobra.Command, args []string) {
```
Comment (campoy) [inline cmd/srcd/cmd/web.go]: This function and the one given to `webParseCmd` are basically the same one, consider unifying.

**Comment by erizocosmico** [comment]
Generalized the start and stop of components, put the code to start components in a single place (so no more repeated component definitions) and unified the code for the two web cmds.

File: cmd/srcd-server/engine/web.go (hunk 1 above)
Comment (campoy) [reply cmd/srcd-server/engine/web.go]: no no, I mean you used `bblfshWebPrivatePort` instead of `gitbaseWebPrivatePort`
I think that's a mistake, is it?

**Comment by campoy** [comment]
Please solve the conflicts and merge.

**Pull request merged**